sniffles/test/data_files/rules/test_rules1.rules

```diff
@@ -8,4 +8,4 @@ alert tcp $EXTERNAL_NET any -> $HOME_NET 8081 (msg:"test1-6"; flow:to_server,est
 alert tcp $EXTERNAL_NET any -> $HOME_NET $HTTP_PORTS (msg:"test1-7"; flow:to_server,established; content:"/struts"; nocase; http_uri; content:"|25|252f"; nocase; http_raw_uri; classtype:attempted-recon; sid:7;)
 alert tcp $EXTERNAL_NET any -> $HOME_NET $HTTP_PORTS (msg:"test1-8"; flow:to_server,established; content:"Proxy-Connection: Keep-Alive"; http_header; content:"Transfer-Encoding: chunked|0D 0A|Content-Length: 40334"; fast_pattern:only; http_header; content:".dll"; http_uri; classtype:attempted-admin; sid:8;)
 alert tcp $EXTERNAL_NET any -> $HOME_NET 8080 (msg:"test1-9"; flow:to_server,established; content:"%2E%2E/"; fast_pattern:only; content:"%2E%2E/"; nocase; http_raw_uri; pcre:"/\/(\\|%5C)%2E%2E\//"; sid:9;)
-alert tcp $EXTERNAL_NET any -> $HOME_NET $HTTP_PORTS (msg:"test1-10"; flow:to_server,established; content:"POST"; http_method; content:"=|27|"; http_client_body; content:"new"; nocase; http_client_body; pcre:"/new(%20)?(javax?|org|com)/i"; classtype:attempted-admin; sid:10;)
\ No newline at end of file
+alert tcp $EXTERNAL_NET any -> $HOME_NET $HTTP_PORTS (msg:"test1-10"; flow:to_server,established; content:"POST"; http_method; content:"=|27|"; http_client_body; content:"new"; nocase; http_client_body; pcre:"/new(%20)?(javax?|org|com)/i"; classtype:attempted-admin; sid:10;)
```

Comment:
The intended way would be to create an HTTP_Stream that extends Traffic_Stream and wrap all necessary functionality in that.  You can use the ScanTraffic traffic stream and its use as a guide.  For the http generation itself, I think you can just use what is there without the need to change any of it (i.e. you can feed it a rule or set of rules that meets your criteria without every having to actually have rules--in other words you could create the rules in the HTTP_Stream object).  I am not saying that it would be easier, but it would be more modular.

The second choice, which should have a user story somewhere, would be to create a script to build rules.  Use that to create a rule-set that meets criteria and then Sniffles doesn't need to change at all.  I know I want a tool like that eventually so that we can have more control over what is generated.  It helps in many evaluation cases.  However, it may be a bit much for this particular case.  Also, to note, a Rule can does not need to define a specific stream.  A single rule can define an arbitrary stream and use a regex as a content generator.  It can define number of packets, etc, but best of all, can be used over and over to create multiple, different requests.